src/plugins/chart_expressions/expression_xy/public/components/xy_chart.tsx

```diff
@@ -505,27 +509,15 @@ export function XYChart({
     ];
 
     if (xySeries.seriesKeys.length > 1) {
-      const pointValue = xySeries.seriesKeys[0];
-      const splitAccessor = layer.splitAccessor
-        ? getAccessorByDimension(layer.splitAccessor, table.columns)
-        : undefined;
-
-      const splitFormat = splitAccessor
-        ? fieldFormats[layer.layerId].splitSeriesAccessors[splitAccessor]
-        : undefined;
-      const splitFormatter = formatFactory(splitFormat);
-
-      points.push({
-        row: table.rows.findIndex((row) => {
-          if (splitAccessor) {
-            if (formattedDatatables[layer.layerId]?.formattedColumns[splitAccessor]) {
-              return splitFormatter.convert(row[splitAccessor]) === pointValue;
-            }
-            return row[splitAccessor] === pointValue;
-          }
-        }),
-        column: table.columns.findIndex((col) => col.id === splitAccessor),
-        value: pointValue,
+      xySeries.splitAccessors.forEach((value, key) => {
+        rowIndex = table.rows.findIndex((row) => {
```

Comment:
According to the same style guidelines, it is necessary to avoid mutations/reassigning of the variables. So, we have to define the variable in the current scope and assign it, or directly assign the value to the field of the object.

Reply:
got it. Done